src/models/team.ts

```diff
@@ -0,0 +1,249 @@
+/*
+Copyright 2020 Balena
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import type * as BalenaSdk from '..';
+import type { InjectedDependenciesParam, PinePostResult, Team } from '..';
+import * as errors from 'balena-errors';
+import { isId, isNotFoundResponse, mergePineOptions } from '../util';
+
+const getTeamModel = function (deps: InjectedDependenciesParam) {
+	const { pine, sdkInstance } = deps;
+
+	/**
+	 * @summary Creates a new Team
+	 * @name create
+	 * @public
+	 * @function
+	 * @memberof balena.models.team
+	 *
+	 * @description This method creates a new team.
+	 *
+	 * @param {Number} organizationSlugOrId - Required: the organization slug or id the team will be part of.
+	 * @param {String} name - Required: the name of the team that will be created.
+	 *
+	 * @fulfil {Object} - Team
+	 * @returns {Promise}
+	 *
+	 * @example
+	 * balena.models.team.create(1239948, 'MyTeam').then(function(team) {
+	 * 	console.log(team);
+	 * });
+	 *
+	 * @example
+	 * balena.models.team.create('myOrgHandle', 'MyTeam')
+	 * .then(function(team) {
+	 *   console.log(team);
+	 * });
+	 */
+	const create = async function (
+		organizationSlugOrId: string | number,
+		name: string,
+	): Promise<PinePostResult<Team>> {
+		const orgId = isId(organizationSlugOrId)
+			? organizationSlugOrId
+			: (await sdkInstance.models.organization.get(organizationSlugOrId))?.id;
```

Comment:
I would be fine if we always did a GET request to confirm that the org exists and give a more helpful error in case it doesn't.
The `isId` in most cases is an optimization for methods that the UI uses in batch actions.